### PR TITLE
FB-324-remove-FOUC

### PR DESCRIPTION
**Pull request by paulinabjacykowska**
Previous investigation showed that what is causing the FOUC is using `styled-components` with `docusaurus` ([here](https://vazco1.atlassian.net/browse/FB-324?focusedCommentId=45331) are two discussions about it linked). 

To prevent this behaviour this PR consists of: 
- removing `styled-components` from all files in `sections` folder
- moving all styles from those files to the `custom.css` file 

To make styles more modular, maintainable, and organised, the CSS `BEM (Block, Element, Modifier)` approach.

Those changes have successfully eliminated FOUC, however, for the purpose of consistency, I think in the future we should remove the usage of styled-components from the remaining files as well.

### CODE REVIEW

File: src/css/custom.css [hunk 1]
```diff
@@ -194,23 +194,450 @@ table {
...
+  cursor: pointer;
+}
+
+.footer-styled__column-button {
+  padding: 0;
+}
+
+.footer-styled__column-link {
+  text-decoration: none !important;
+}
```
Comment (mkaliszewski) [inline src/css/custom.css:545]: Not needed at all. Without it, it fallbacks to `main a`

Screenshot from 2023-08-09 10-15-50

File: src/css/custom.css [hunk 2]
```diff
@@ -78,13 +78,13 @@ table {
 /* Make "Buy Now" link styled as button */
 .buy-now-nav-item {
   outline: none;
-  background: var(--theme-color-dark-navy-blue) !important;
+  background: var(--theme-color-stormy-blue) !important;
```
Comment (mkaliszewski) [inline src/css/custom.css:81]: We can remove `!important`

File: src/css/custom.css [hunk 3]
```diff
@@ -96,8 +96,8 @@ table {
 
 .buy-now-nav-item:hover {
   background: var(--theme-color-white) !important;
-  color: var(--theme-color-dark-navy-blue) !important;
-  border: 2px solid var(--theme-color-dark-navy-blue) !important;
+  color: var(--theme-color-stormy-blue) !important;
+  border: 2px solid var(--theme-color-stormy-blue) !important;
```
Comment (mkaliszewski) [inline src/css/custom.css:100]: `!important` needed only for `color`

File: src/css/custom.css [hunk 4]
```diff
@@ -194,23 +194,450 @@ table {
...
+  padding: 0;
+}
+
+.footer-styled__column-link {
+  text-decoration: none !important;
+}
+
+.footer-styled__column-mail {
+  text-decoration: none !important;
+}
```
Comment (mkaliszewski) [inline src/css/custom.css:549]: Not needed at all. Without it, it fallbacks to `main a`

**Review by mkaliszewski — APPROVED** [review]
NB comments

File: src/css/custom.css (hunk 2 above)
Comment (paulinabjacykowska) [reply src/css/custom.css:81]: If we remove it it doesn't work properly for mobile view :( 

https://github.com/vazco/forminer/assets/89641881/9c21572e-0a87-48ba-a48c-fb0569b702a6

File: src/css/custom.css (hunk 1 above)
Comment (paulinabjacykowska) [reply src/css/custom.css:545]: If I remove it or remove `!important` it has an underline 🙈

https://github.com/vazco/forminer/assets/89641881/35643cdf-3f21-43eb-9d1c-f9d695b2d3ee

https://github.com/vazco/forminer/assets/89641881/f0b5a9c1-a4f0-4aed-9bd3-0bfa4843e9c8

File: src/css/custom.css (hunk 4 above)
Comment (paulinabjacykowska) [reply src/css/custom.css:549]: Same as in https://github.com/vazco/forminer/pull/62#discussion_r1288187776

File: src/css/custom.css (hunk 2 above)
Comment (mkaliszewski) [reply src/css/custom.css:81]: My bad, I haven't checked it!

File: src/css/custom.css (hunk 3 above)
Comment (mkaliszewski) [reply src/css/custom.css:100]: Okay, now I see if we want to delete it then we need to add another !important somewhere else :dizzy:

File: src/css/custom.css [hunk 5]
```diff
@@ -194,23 +194,450 @@ table {
...
+
+.footer-styled__showcase-card-text {
+  text-align: center;
+  padding-top: 1.5em;
+  font-size: 16px;
+}
+
+.footer-styled__column-item {
+  margin-bottom: 10;
+}
```
Comment (mkaliszewski) [inline src/css/custom.css:523]: I just noticed there is no unit

File: src/css/custom.css (hunk 1 above)
Comment (paulinabjacykowska) [reply src/css/custom.css:545]: Sure, it's a great idea!

**Pull request merged**